### ISSUE TITLE
Creating a group with `members` attribute

### ISSUE THREAD

**Issue by jargoud** [limosa-io/laravel-scim-server#98]
Hello,

Let say we have this payload:

```json
{
  "schemas": [
    "urn:ietf:params:scim:schemas:core:2.0:Group"
  ],
  "displayName": "Administrators",
  "members": [
    {
      "value": "1",
      "display": "john.doe@example.com"
    }
  ]
}
```

Creating such group provokes a server error, due to `members` attribute and the use of `MutableCollection` object.

[In `ResourceController`](https://github.com/limosa-io/laravel-scim-server/blob/master/src/Http/Controllers/ResourceController.php#L66):

```php
public static function createFromSCIM($resourceType, $input, PolicyDecisionPoint $pdp = null, Request $request = null, $allowAlways = false, $isMe = false)
{
    // …

    $resourceObject = $resourceType->getFactory()();

    $resourceType->getMapping()->replace($input, $resourceObject); // <=== HERE

    //validate
    $newObject = Helper::flatten(Helper::objectToSCIMArray($resourceObject, $resourceType), $resourceType->getSchema());

    $flattened = self::validateScim($resourceType, $newObject, $resourceObject);

    $resourceObject->save();

    return $resourceObject;
}
```

Instruction `$resourceType->getMapping()->replace($input, $resourceObject);` will call [`replace` method from `MutableCollection`](https://github.com/limosa-io/laravel-scim-server/blob/master/src/Attribute/MutableCollection.php#L61), which will then call `sync` method from relation: `$object->{$this->attribute}()->sync($existingObjects->all());`.

This will (probably) lead (it happens in my case) to a database exception, because `$resourceObject` is actually not persisted yet, so we cannot use its primary key as foreign key in the pivot table.

Trying to save the model before syncing will lead to a uniqueness exception, because a group with the given name already exists, when the time to validate data comes.

Is there any workaround?

Thanks,

Jeremy

**Comment by arietimmerman** [comment]
It took a while, but I may have a fix for this issue. 

See https://github.com/limosa-io/laravel-scim-server/commit/f4c2af52d2b7038c4208ab314965e80fd6466373

I'll do a bit of extra checking, and after that, I'll merge it in.

The way I have fixed it now may have some performance impact, do to the use of the `load` method. But perhaps that's not the case or acceptable.

**Comment by arietimmerman** [comment]
Solved via #100

**Comment by arietimmerman** [comment]
See it live in action here: https://scim.dev/playground/groups.html#create-group-with-members
